src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs

```diff
@@ -50,7 +53,10 @@ public List(int capacity)
             if (capacity == 0)
                 _items = s_emptyArray;
             else
-                _items = new T[capacity];
+            // _items = new T[capacity];
+
+            // DO NOT MERGE: for testing only 
```

Comment:
ArrayPool is where this API should be used for real.

Reply:
yeah, that's what we were planning to do.

Reply:
Yeah. These are just very popular types giving a good variety of test coverage until we have real tests.

Reply:
I had some nice bug finds from this. The zeroing is fairly delicate in edge cases.

---
In reply to: [277937587](https://github.com/dotnet/coreclr/pull/24096#discussion_r277937587) [](ancestors = 277937587)

Reply:
I assume any actual use - in ArrayPool or anywhere else will be separate PRs.

---
In reply to: [277933656](https://github.com/dotnet/coreclr/pull/24096#discussion_r277933656) [](ancestors = 277933656)

Reply:
If you are adding this as internal API, you should add one real use of this in the same PR (e.g. the array pool one). We try to avoid PRs that add dead code.